src/components/Match/WardLog.jsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { threshold, formatSeconds, extractTransitionClasses } from 'utility';
+import { Row, Col } from 'react-flexbox-grid';
+// import heroes from 'dotaconstants/json/heroes.json';
+import Fixed from 'utility/components/Fixed';
+import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
+import strings from 'lang';
+
+import { heroTd } from './matchColumns';
+import styles from './Match.css';
+
+
+const durationObserverColor = threshold(0, [121, 241, 500], [styles.red, styles.yelor, styles.green]);
+
+const RowItem = ({ style, row, match }) => {
+  const wardKiller = (row.left && row.left.player1) ? heroTd(match.players[row.left.player1]) : '-';
+  const duration = row.left ? row.left.time - row.entered.time : '-';
+  const durationColor = row.type === 'observer' ? durationObserverColor(duration) : 'inherit';
+  const rowStyle = { ...style,
+    backgroundColor: row.key % 2 === 0
+                   ? styles.wardLogRowEvenSurfaceColor
+                   : styles.wardLogRowOddSurfaceColor,
+  };
+  return (
+    <li>
+      <Row component="li" className={styles['ward-log-item']} style={rowStyle} middle="xs">
+        <Col xs={1}>
+          <img height="29" src={`${API_HOST}/apps/dota2/images/items/ward_${row.type}_lg.png`} role="presentation" />
+        </Col>
+        <Col xs>
+          {heroTd(match.players[row.player])}
+        </Col>
+        <Col xs={1} className={styles.timespan}>{formatSeconds(row.entered.time)}</Col>
+        <Col xs={1} className={styles.timespan}>{formatSeconds(row.left && row.left.time) || '-'}</Col>
+        <Col xs={1} className={styles.timespan} style={{ color: durationColor }}>{formatSeconds(duration)}</Col>
+        <Col xs>{wardKiller}</Col>
+      </Row>
+    </li>
+  );
+};
+const PureRowItem = Fixed(RowItem);
+
+const WardLog = (props) => {
+  const columns = ['type', 'owner', 'entered_at', 'left_at', 'duration', 'killed_by'].map(h => `ward_log_${h}`);
```

Comment:
can't we declare this outside of the component to prevent this executing every render?